openmetadata-service/src/main/java/org/openmetadata/service/apps/bundles/searchIndex/ElasticSearchIndexSink.java

```diff
@@ -113,101 +113,240 @@ public void write(List<?> entities, Map<String, Object> contextData) throws Sear
+          .bulkAsync(
+              bulkRequest,
+              RequestOptions.DEFAULT,
+              new ActionListener<BulkResponse>() {
+                @Override
+                public void onResponse(BulkResponse response) {
+                  try {
+                    for (int i = 0; i < response.getItems().length; i++) {
+                      BulkItemResponse itemResponse = response.getItems()[i];
+                      if (itemResponse.isFailed()) {
+                        String failureMessage = itemResponse.getFailureMessage();
+                        String entityData = requests.get(i).toString();
+                        entityErrorList.add(
+                            new EntityError().withMessage(failureMessage).withEntity(entityData));
+                        LOG.warn("Bulk item failed: {}", failureMessage);
+                      }
+                    }
 
-    while (attempt <= maxRetries) {
-      try {
-        semaphore.acquire();
-        try {
-          BulkResponse response = client.bulk(bulkRequest, RequestOptions.DEFAULT);
-          entityErrorList.addAll(getErrorsFromBulkResponse(response));
-          break; // Success, exit retry loop
-        } finally {
-          semaphore.release();
-        }
-      } catch (IOException e) {
-        if (isRetriableException(e)) {
-          attempt++;
-          LOG.warn(
-              "Bulk request failed with retriable exception, retrying attempt {}/{}",
-              attempt,
-              maxRetries);
-          sleepWithBackoff(backoffMillis);
-          backoffMillis = Math.min(backoffMillis * 2, maxBackoffMillis);
-        } else {
-          LOG.error("Bulk request failed with non-retriable exception", e);
-          throw new SearchIndexException(createIndexingError(requests.size(), e));
-        }
-      } catch (InterruptedException e) {
-        Thread.currentThread().interrupt();
-        LOG.error("Bulk request interrupted", e);
-        throw new SearchIndexException(createIndexingError(requests.size(), e));
-      } catch (ElasticsearchException e) {
-        if (isRetriableStatusCode(e.status())) {
-          attempt++;
-          LOG.warn(
-              "Bulk request failed with status {}, retrying attempt {}/{}",
-              e.status(),
-              attempt,
-              maxRetries);
-          sleepWithBackoff(backoffMillis);
-          backoffMillis = Math.min(backoffMillis * 2, maxBackoffMillis);
-        } else {
-          LOG.error("Bulk request failed with non-retriable status {}", e.status(), e);
-          throw new SearchIndexException(createIndexingError(requests.size(), e));
-        }
-      }
+                    int success = response.getItems().length;
+                    int failed = 0;
+                    for (BulkItemResponse item : response.getItems()) {
+                      if (item.isFailed()) {
+                        failed++;
+                      }
+                    }
```

Comment:
can we do something like

```java
int failed = entityErrorList.size()
```

I believe this prevents us from looping twice over the response